docs/configuration-discovery-advanced.md

```diff
@@ -0,0 +1,79 @@
+# Configuration: advanced discovery
+
+The Orchestrator maintains an internal queue of instances to discover. When the instance is scheduled to be discovered, it is added to the queue.
+The queue is consumed by discovery workers. The number of workers is configured via `DiscoveryMaxConcurrency` in a configuration file. This parameter configures how many discoveries can happen in parallel.
+To maintain the up-to-date view of the cluster, the Orchestrator has to monitor all instances using the above-described mechanism periodically. `InstancePollSeconds` configuration parameter says how often the Orchestrator should refresh the information.
+
+In the case when a lot of instances become inaccessible or unhealthy in the manner that examining them takes a long time and finishes with failure discovery workers may be busy mostly with these instances. At the same time healthy ones wait in a discovery queue and are not checked with the desired frequency. It may cause the Orchestrator to loose the proper view of the cluster.
```

Comment:
The Orchestrator may lose the proper view of the cluster when examining a large number of instances, which can take a long time and end in failure. These instances may become inaccessible or unhealthy. Discovery workers may be busy with these instances. Healthy instances wait in the queue and they are not checked in a timely manner.